### DISCUSSION TITLE
How do you get the email provided in a payment to send the invoice?

### DISCUSSION THREAD

**Discussion by Scolymus** [python-telegram-bot/python-telegram-bot#3099] [Q&A] [answered]
I need to send to the user the digital set bought plus the invoice. That's why I put need_email=true in line 104 https://github.com/python-telegram-bot/python-telegram-bot/blob/master/examples/paymentbot.py However, at the end of the process, in line 141 I want to send the product and the invoice via email. How can I recover the email writen by the user?

**Comment by harshil21** [comment, marked as answer]
In the [callback](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/examples/paymentbot.py#L138) which is run when a message containing a successful payment has arrived, you can access the email the user specified by doing: `update.effective_message.successful_payment.order_info.email`

The docs: 
Message - https://docs.python-telegram-bot.org/en/latest/telegram.message.html#telegram.Message.params.successful_payment
SuccessfulPayment - https://docs.python-telegram-bot.org/en/latest/telegram.successfulpayment.html
OrderInfo - https://docs.python-telegram-bot.org/en/latest/telegram.orderinfo.html#telegram.OrderInfo
